### PR TITLE
Incorporate blank non blank

### PR DESCRIPTION
**Pull request by r-b-g-b**
Scaffolding for incorporating blank/non-blank detection as a first step in the prediction pipeline.

### CODE REVIEW

**Comment by r-b-g-b** [comment]
@pjbull Made the changes we discussed today, should be ready to go!

File: zamba/models/cnnensemble_model.py [hunk 1]
```diff
@@ -142,24 +152,38 @@ def _find_non_non_blank_frames(self):
...
+            blank_probabilities,
+            index=[str(file_name) for file_name in file_names],
+            name="blank",
+        )
+
+        blank_file_names = blank_probabilities.loc[blank_probabilities > threshold].index
+        blank_file_names = set(Path(file_name) for file_name in blank_file_names)
 
-        return blank_videos
+        return blank_file_names, blank_probabilities
```
Comment (pjbull) [inline zamba/models/cnnensemble_model.py]: If you return `blank_probabilities.to_frame()`, it should make the join etc. easier.

File: zamba/models/cnnensemble_model.py [hunk 2]
```diff
@@ -89,6 +93,10 @@ def predict(self, file_names):
 
             all_skipped.update({str(f) for f in skipped})
 
+        blank_probabilities = blank_probabilities.loc[
+            ~blank_probabilities.index.isin(all_skipped)
+        ]
```
Comment (pjbull) [inline zamba/models/cnnensemble_model.py]: With the `to_frame` below, we can remove these lines.

File: zamba/models/cnnensemble_model.py [hunk 3]
```diff
@@ -106,9 +114,11 @@ def predict(self, file_names):
             columns=config.CLASSES,
         )
 
-        blanks = pd.DataFrame(0, index=blank_file_names, columns=preds.columns)
-        blanks["blank"] = 1
-        preds = pd.concat([preds, blanks], axis=0)
+        if "blank" in preds:
+            del preds["blank"]
```
Comment (pjbull) [inline zamba/models/cnnensemble_model.py]: With the `to_frame` below, we can remove these lines.

File: zamba/models/cnnensemble_model.py [hunk 4]
```diff
@@ -106,9 +114,11 @@ def predict(self, file_names):
...
+
+        preds = preds.join(blank_probabilities, how="outer")
```
Comment (pjbull) [inline zamba/models/cnnensemble_model.py]: With `to_frame`, I think we can do `preds = blank_probabilities.join(preds.drop('blank', axis=1)`

File: zamba/models/cnnensemble_model.py [hunk 5]
```diff
@@ -67,14 +68,17 @@ def predict(self, file_names):
...
         Returns:
-
+            pd.DataFrame: A table of class probabilities, where index is the file name and columns are the different
+            classes
         """
-        blank_file_names = self._find_blank_videos(file_names)
-        self.logger.debug(
-            "Blank file names %s", "\n".join([str(file_name) for file_name in blank_file_names])
+        blank_file_names, blank_probabilities = self._find_blank_videos(file_names, threshold=0.5, dummy=True)
+        self.logger.debug("Blank file names %s", str(blank_file_names))
```
Comment (pjbull) [inline zamba/models/cnnensemble_model.py]: I prefer `f""` even for log messages

**Comment by pjbull** [comment]
@r-b-g-b Can you also rebase? That should make appveyor happier

**Comment by pjbull** [comment]
@r-b-g-b can you open a PR for the appveyor failures?

**Pull request merged**